Assignment4/MyLibrary/MyLibrary/SetWriter.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.IO;
+
+namespace MyLibrary.Writer
+{
+    public sealed class SetWriter : IDisposable
+    {
+        private bool _DisposedValue;
+
+        private string? _filePath;
+
+        public string? FilePath
+        {
+            get => _filePath;
+            set => _filePath = value ??
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        private StreamWriter? fileWriter;
+
+        public StreamWriter? FileWriter
+        {
+            get => fileWriter;
+            set => fileWriter = value ?? 
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        public SetWriter(string filePath)
+        {
+            FileWriter = new StreamWriter(filePath);
+        }
+
+        public void WriteSet(NumSet numSet)
+        {
+            FileWriter!.Write(numSet.ToString());
```

Comment:
```suggestion
            FileWriter.Write(numSet.ToString());
```
Use the `!` cautiously. If you find that you are using it often there is likely something wrong with how you are writing your code. 

Also, you may consider using `WriteLine` here.